infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java

```diff
@@ -115,6 +127,42 @@ protected AbstractAddonInfrastructure() {
         instancesWithoutNodesMap = new HashMap<>();
     }
 
+    @Override
+    protected void notifyAcquiredNode(Node node) throws RMException {
+        incrementNumberOfAcquiredNodesWithLockAndPersist();
+    }
+
+    private void incrementNumberOfAcquiredNodesWithLockAndPersist() {
+        setPersistedInfraVariable(() -> {
+            int updated = (int) this.persistedInfraVariables.get(NB_ACQUIRED_NODES_KEY) + 1;
```

Comment:
Have you consider storing allocated node numbers in AtomicInteger attribute ? It could limit the interaction with the persisted variables to during increment and decrement operations.

Reply:
Done, thanks.